sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubProducerClientOptions.cs

```diff
@@ -20,6 +21,31 @@ public class EventHubProducerClientOptions
         /// <summary>The set of options to govern retry behavior and try timeouts.</summary>
         private EventHubsRetryOptions _retryOptions = new EventHubsRetryOptions();
 
+        /// <summary>
+        ///   Indicates whether or not the producer should enable idempotent publishing to the Event Hub partitions.  If
+        ///   enabled, the producer will only be able to publish directly to partitions; it will not be able to publish to
+        ///   the Event Hubs gateway for automatic partition routing nor using a partition key.
+        /// </summary>
+        ///
+        /// <value><c>true</c> if the producer should enable idempotent partition publishing; otherwise, <c>false</c>.</value>
+        ///
+        public bool EnableIdempotentPartitions { get; set; }
+
+        /// <summary>
+        ///   The set of options that can be specified to influence publishing behavior specific to the configured Event Hub partition.  These
+        ///   options are not necessary in the majority of scenarios and are intended for use with specialized scenarios, such as when
+        ///   recovering the state used for idempotent publishing.
+        ///
+        ///   <para>It is highly recommended that these options only be specified if there is a proven need to do so; Incorrectly configuring these
+        ///   values may result in an <see cref="EventHubProducerClient" /> instance that is unable to publish to the Event Hubs.</para>
+        /// </summary>
+        ///
+        /// <remarks>
+        ///   These options are ignored when publishing to the Event Hubs gateway for automatic routing or when using a partition key.
+        /// </remarks>
+        ///
+        public Dictionary<string, PartitionPublishingOptions> PartitionOptions { get; } = new Dictionary<string, PartitionPublishingOptions>();
```

Comment:
Is PartitionOptions too generic a name? If someday we have other options for every partition, this name sounds it has all options for partitions. But its type is PartitionPublishingOptions.

Reply:
These are attached to the producer client, so there's an implied context; I don't see value in making the property name more complicated.   As you noted, the type also provides an explicit context.